logging/proto_conversions.go

```diff
@@ -0,0 +1,81 @@
+package logging
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.viam.com/utils/protoutils"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+// FieldToProto converts a zap.Field to a *structpb.Struct.
+func FieldToProto(field zap.Field) (*structpb.Struct, error) {
+	// Zap encodes float64s with very large int64s. Proto conversions have some
+	// loss with very large int64s. float32s are also encoded with int64s, but
+	// the int64 encodings are not large enough to cause loss in conversion.
+	// See https://pkg.go.dev/google.golang.org/protobuf@v1.32.0/types/known/structpb#NewValue.
+	//
+	// Use a hacky combination of fmt and math to store float64s as strings.
+	if field.Type == zapcore.Float64Type {
+		field.String = fmt.Sprintf("%f", math.Float64frombits(uint64(field.Integer)))
+	}
+
+	return protoutils.StructToStructPb(field)
+}
+
+// FieldKeyAndValueFromProto examines a *structpb.Struct and returns its key
+// string and native golang value.
+func FieldKeyAndValueFromProto(field *structpb.Struct) (string, any, error) {
```

Comment:
If these are module-logging specific, we should name them as such or add a comment.

For example, this means the cli can no longer use this function because it is not compatible with the app-fields-float64-in-integer storage mechanism

Reply:
Though cli could use this if you returned a value that tried to parse the integer field (instead of returning err)

Reply:
I would like the CLI to use it; I'll move the swallowing of the error into this function.